effekt/shared/src/main/scala/effekt/jit/PrettyPrinter.scala

```diff
@@ -54,15 +57,13 @@ object PrettyPrinter extends ParenPrettyPrinter {
         "frameDescriptor" -> toDoc(frameDescriptor),
         "instructions" -> jsonList(instructions.map(toDoc) ++ List(toDoc(terminator)))
       ))
-    } // TODO: throw Error("Internal error: Unnumbered BasicBlock")
+    }
   }
 
   def toDoc(frameDescriptor: FrameDescriptor): Doc = {
-    jsonObjectSmall(ListMap( // TODO: calculate befor pretty-printing
-      "regs_int" -> frameDescriptor.locals.applyOrElse(RegisterType.Integer, x => 0).toString,
-      "regs_cont" -> frameDescriptor.locals.applyOrElse(RegisterType.Continuation, x => 0).toString,
-      "regs_adt" -> frameDescriptor.locals.applyOrElse(RegisterType.Datatype, x => 0).toString
-    ))
+    jsonObjectSmall(RegisterType.valuesNonErased.map(ty => ("regs_%s".format(ty.name)) ->
+      (frameDescriptor.locals.getOrElse(ty, 0).toString: Doc)
```

Comment:
Should this really be `getOrElse` 0? Or should it rather be apply and crash if not present?

Reply:
I retract. Sorry. This is for each register type not a single register.